tests/csep1_tests.py

```diff
@@ -0,0 +1,336 @@
+    """
+        
+    return scipy.stats.poisson.ppf(random_matrix, forecast) 
+
+
+def l_test(observation, forecast, num_simulations=1000, model_name='None'):
+    """
+    Computes L test for Observed and Forecasted Catalogs
+    
+    We find the Joint Log Likelihood between Observed Catalog and Forecasted 
+    Catalogs. Later on, "num_simulations" Forecast Realizations are generated using Forecasted Catalog and Random numbers through inverse poission CDF.
+    Then Joint Likelihood of Forecast and Forecast Realizations are computed. 
+    Actual Joint Log Likelihood and Simulated Joint Log Likelihoods are then employed to compute Quantile Score
+    
+    Args
+    observation:Observed (Grided) seismicity (Numpy Array):
+                An Observation has to be Number of Events in Each Bin
+                It has to be a either zero or positive integer only (No Floating Point)
+    forecast:   Forecast of a Model (Grided) (Numpy Arrazy)
+                A forecast has to be in terms of Average Number of Events in Each Bin 
+                It can be anything greater than zero
+    num_simulation: Number of simulated Catalogs to be generated (Non-negative Integer)
+                A non-negative integer indicating the number of realized forecasts to be generated
+    model_name: The chosen name of model.             
+    
+    Returns
+    A dictionary of 4 elements
+                {'quantile' : Quantile Score of the Model,
+                    'll_actual': joint Log Likelihood of Observation and Actual Forecast,
+                     'll_simulation': Numpy array of Joint Likelihood of Simulated Forecasts 
+                      'model_name': The name of the model, provided in the input}
+    """
+                                                #, seed=None
+    if not isinstance(observation, numpy.ndarray):
+        raise TypeError('observation must be numpy.ndarray')
+    if not isinstance(forecast, numpy.ndarray):
+        raise TypeError('forecast must be numpy.ndarray')
+        
+
+    ll_actual = numpy.sum(poisson_log_likelihood(observation, forecast))   
+
+    ll_simulations = numpy.zeros(num_simulations)
+    
+    # compute simulated log-likelihoods
+    for count in range(num_simulations):
+        # random percentile for each independent bin
+        random_matrix = numpy.random.random(forecast.shape) 
+        # computes the inverse cdf of the poisson random variable
+        forecast_realization = _forecast_realization_inverse_cdf(random_matrix, forecast)  
+        ll_simulations[count] = numpy.sum(poisson_log_likelihood(forecast_realization, forecast))
+    quantile_score = numpy.sum(ll_simulations <= ll_actual) / num_simulations #Apply 3rd UNIT TEST here on Quantile Score. 
+    
+
+    L_Test_Eval = { 'model_name': model_name,
+                    'quantile' : quantile_score,
+                    'll_actual': ll_actual,
+                     'll_simulation':ll_simulations}
```

Comment:
we can use EvaluationResult

result = EvaluationResult(test_distribution=This is the ll_simulations
                          name=Name of forecast,
                          observed_statistic=<ll_actual>,
                          quantile=(delta_1, delta_2),
                          status='Normal',
                          obs_catalog_repr=obs.date_accessed,
                          sim_name=model_name,
                          min_mw=None,
                          obs_name=None)